obspy/core/trace.py

```diff
@@ -2779,6 +3241,207 @@ def _data_sanity_checks(value):
                "arrays are allowed for initialization.") % str(value.shape)
         raise ValueError(msg)
 
+class FrequencyDomainTrace(BaseTrace):
```

Comment:
I would move the whole class to a separate file, maybe call it `frequency_domain_trace.py` or something along this line? Small files are easier to work with and this is IMHO a sensible split.